src/components/Form/FormFieldError.tsx

```diff
@@ -1,36 +1,87 @@
-import { ComponentPropsWithoutRef, ElementRef, forwardRef } from 'react';
+import { ElementRef, ReactNode, useContext } from 'react';
 
-import { FormErrorMessage, FormHelperText, SlideFade } from '@chakra-ui/react';
+import { Flex, FlexProps, SlideFade } from '@chakra-ui/react';
+import {
+  ControllerProps,
+  FieldError,
+  FieldPath,
+  FieldValues,
+  useFormContext,
+} from 'react-hook-form';
 import { LuAlertCircle } from 'react-icons/lu';
 
+import {
+  FormFieldControllerContext,
+  FormFieldControllerContextValue,
+} from '@/components/Form/FormFieldController';
 import { Icon } from '@/components/Icons';
+import { fixedForwardRef } from '@/lib/utils';
 
-import { useFormField, useFormFieldContext } from './FormField';
+type FormFieldErrorProps<
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>,
+> = Omit<FlexProps, 'children'> & {
+  children?: (params: { error?: FieldError }) => ReactNode;
+} & (
+    | Required<Pick<ControllerProps<TFieldValues, TName>, 'control' | 'name'>>
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    | {}
+  );
 
-export const FormFieldError = forwardRef<
-  ElementRef<typeof FormErrorMessage>,
-  ComponentPropsWithoutRef<typeof FormErrorMessage>
->(({ children, ...props }, ref) => {
-  const { error } = useFormField();
-  const ctx = useFormFieldContext();
+const FormFieldErrorComponent = <
+  TFieldValues extends FieldValues = FieldValues,
+  TName extends FieldPath<TFieldValues> = FieldPath<TFieldValues>,
+>(
+  { children, ...props }: FormFieldErrorProps<TFieldValues, TName>,
+  ref: ElementRef<typeof Flex>
+) => {
+  const ctx = useContext<FormFieldControllerContextValue<
+    TFieldValues,
+    TName
+  > | null>(FormFieldControllerContext as ExplicitAny);
+  const { formState } = useFormContext<TFieldValues, TName>();
+  const control = 'control' in props ? props.control : ctx?.control;
+  const name = 'name' in props ? props.name : ctx?.name;
```

Comment:
```suggestion
  const control = props?.control ?? ctx?.control;
  const name = props?.name ?? ctx?.name;
```

Is something like this okay ?

Reply:
I'm okay with that change but Typescript not 😅